website/docs/r/stack_hci_extension.html.markdown

```diff
@@ -0,0 +1,103 @@
+---
+subcategory: "Azure Stack HCI"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_stack_hci_extension"
+description: |-
+  Manages an Azure Stack HCI Extension.
+---
+
+# azurerm_stack_hci_extension
+
+Manages an Azure Stack HCI Extension.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example-hci-ext"
+  location = "West Europe"
+}
+
+resource "azurerm_log_analytics_workspace" "example" {
+  name                = "example-hci-law"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  sku                 = "PerGB2018"
+  retention_in_days   = 30
+}
+
+resource "azurerm_stack_hci_extension" "example" {
+  name                               = "example-shce"
+  arc_setting_id                     = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/example-hci/providers/Microsoft.AzureStackHCI/clusters/hci-cl/arcSettings/default"
+  publisher                          = "Microsoft.EnterpriseCloud.Monitoring"
+  type                               = "MicrosoftMonitoringAgent"
+  auto_upgrade_minor_version_enabled = true
+  automatic_upgrade_enabled          = false
+  type_handler_version               = "1.22.0"
+
+  protected_settings = <<PROTECTED_SETTINGS
+{
+	"workspaceKey": "${azurerm_log_analytics_workspace.test.primary_shared_key}"
+}
+PROTECTED_SETTINGS
+
+  settings = <<SETTINGS
+{
+	"workspaceId": "${azurerm_log_analytics_workspace.test.workspace_id}"
+}
+SETTINGS
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name which should be used for this Azure Stack HCI Extension. Changing this forces a new resource to be created.
+
+* `arc_setting_id` - (Required) The ID of the Azure Stack HCI Cluster Arc Setting. Changing this forces a new resource to be created.
+
+* `publisher` - (Required) The name of the extension handler publisher, such as `Microsoft.Azure.Monitor`. Changing this forces a new resource to be created.
+
+* `type` - (Required) Specifies the type of the extension. For example `CustomScriptExtension` or `AzureMonitorLinuxAgent`. Changing this forces a new resource to be created.
+
+* `auto_upgrade_minor_version_enabled` - (Optional) Indicates whether the extension should use a newer minor version if one is available at deployment time. Once deployed, however, the extension will not upgrade minor versions unless redeployed, even with this property set to true. Possible values are `true` and `false`.
```

Comment:
what are the differences between `auto_upgrade_minor_version_enabled` and `automatic_upgrade_enabled`?